src/main/java/org/kiwiproject/jersey/client/RegistryAwareClient.java

```diff
@@ -28,9 +34,32 @@
     @Getter(AccessLevel.PACKAGE)
     private final RegistryClient registryClient;
 
+    /**
+     * Creates a new {@link RegistryAwareClient} with the given {@link Client} and {@link RegistryClient}. This method
+     * will NOT attach any headers to the request automatically.
+     *
+     * @param client            the Jersey client to use
+     * @param registryClient    the registry lookup client
+     */
     public RegistryAwareClient(Client client, RegistryClient registryClient) {
+        this(client, registryClient, null);
+    }
+
+    /**
+     * Creates a new {@link RegistryAwareClient} with the given {@link Client}, {@link RegistryClient}
+     * and {@link Supplier} that will be used to automatically attach request headers.
+     *
+     * @param client            the Jersey client to use
+     * @param registryClient    the registry lookup client
+     * @param headersSupplier   a supplier to retrieve headers to attach to the request
+     */
+    public RegistryAwareClient(Client client, RegistryClient registryClient, Supplier<Map<String, Object>> headersSupplier) {
```

Comment:
So how does one get this supplier in here if you're using `RegistryAwareClientBuilder`?

Reply:
Fixed by adding support in `ClientBuilder` (and in the `RegistryAwareClientBuilder` interface)